core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -244,14 +243,15 @@ public static void updateTableInfo(TableInfo tableInfo) {
   public static CarbonTable buildTable(
       String tablePath,
       String tableName,
-      Configuration configuration) throws IOException {
-    TableInfo tableInfoInfer = CarbonUtil.buildDummyTableInfo(tablePath, "null", "null");
-    CarbonFile carbonFile = getLatestIndexFile(FileFactory.getCarbonFile(tablePath, configuration));
-    if (carbonFile == null) {
-      throw new RuntimeException("Carbon index file not exists.");
+      Configuration configuration,
+      boolean isFile) throws IOException {
+    if (isFile) {
+      int index = tablePath.lastIndexOf("/");
```

Comment:
This changes are not related to Binary, can we raise in a separate PR later ?

Reply:
ok